### ISSUE TITLE
Pygame-1.9.6 test failures with SDL-1.2 tip

### ISSUE THREAD

**Issue by cgohlke** [pygame/pygame#1100]
Just reporting: I get the following test failures using pygame-1.9.6 with SDL-1.2.tip (a936f9bd3e38) on Windows (any Python version, 32 and 64-bit). Basically `pygame.display.set_mode()` fails with `depth` 8, 16, and 24. 

```
pygame 1.9.6
Hello from the pygame community. https://www.pygame.org/contribute.html
skipping pygame.tests.cdrom_test (tag 'interactive')
skipping pygame.tests.midi_test (tag 'interactive')
skipping pygame.tests.touch_test (tag 'SDL1_ignore')
loading pygame.tests.base_test
loading pygame.tests.blit_test
loading pygame.tests.bufferproxy_test
loading pygame.tests.camera_test
loading pygame.tests.color_test
loading pygame.tests.compat_test
loading pygame.tests.constants_test
loading pygame.tests.cursors_test
loading pygame.tests.display_test
loading pygame.tests.draw_test
loading pygame.tests.event_test
loading pygame.tests.fastevent_test
loading pygame.tests.font_test
loading pygame.tests.freetype_test
loading pygame.tests.ftfont_test
loading pygame.tests.gfxdraw_test
loading pygame.tests.image__save_gl_surface_test
loading pygame.tests.image_test
loading pygame.tests.imageext_test
loading pygame.tests.joystick_test
loading pygame.tests.key_test
loading pygame.tests.mask_test
loading pygame.tests.math_test
loading pygame.tests.mixer_music_test
loading pygame.tests.mixer_test
loading pygame.tests.mouse_test
loading pygame.tests.overlay_test
loading pygame.tests.pixelarray_test
loading pygame.tests.pixelcopy_test
loading pygame.tests.rect_test
loading pygame.tests.rwobject_test
loading pygame.tests.scrap_test
loading pygame.tests.sndarray_test
loading pygame.tests.sprite_test
loading pygame.tests.surface_test
loading pygame.tests.surfarray_test
loading pygame.tests.surflock_test
loading pygame.tests.sysfont_test
loading pygame.tests.threads_test
loading pygame.tests.time_test
loading pygame.tests.transform_test
loading pygame.tests.version_test
..........................................................................................................................xxxx......EEEEEEEEEEEE.............ssssssEEEEEEEE............................................................................................................................................x.x....................................s.....s.........xx...............................................................................................................................................................................................................s..........................................................................................................................................................s........................................................................................................................................................................................s.......s............................................s......................................s
======================================================================
ERROR: test_aaline__color (pygame.tests.draw_test.DrawLineTest)
Tests if the aaline drawn is the correct color.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "X:\Python37\lib\site-packages\pygame\tests\draw_test.py", line 453, in setUp
    surface = pygame.display.set_mode(size, flags, depth)
pygame.error

======================================================================
ERROR: test_aaline__gaps (pygame.tests.draw_test.DrawLineTest)
Tests if the aaline drawn contains any gaps.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "X:\Python37\lib\site-packages\pygame\tests\draw_test.py", line 453, in setUp
    surface = pygame.display.set_mode(size, flags, depth)
pygame.error

======================================================================
ERROR: test_aaline_endianness (pygame.tests.draw_test.DrawLineTest)
test color component order
----------------------------------------------------------------------
Traceback (most recent call last):
  File "X:\Python37\lib\site-packages\pygame\tests\draw_test.py", line 453, in setUp
    surface = pygame.display.set_mode(size, flags, depth)
pygame.error

======================================================================
ERROR: test_aalines__color (pygame.tests.draw_test.DrawLineTest)
Tests if the aalines drawn are the correct color.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "X:\Python37\lib\site-packages\pygame\tests\draw_test.py", line 453, in setUp
    surface = pygame.display.set_mode(size, flags, depth)
pygame.error

======================================================================
ERROR: test_aalines__gaps (pygame.tests.draw_test.DrawLineTest)
Tests if the aalines drawn contain any gaps.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "X:\Python37\lib\site-packages\pygame\tests\draw_test.py", line 453, in setUp
    surface = pygame.display.set_mode(size, flags, depth)
pygame.error

======================================================================
ERROR: test_color_validation (pygame.tests.draw_test.DrawLineTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "X:\Python37\lib\site-packages\pygame\tests\draw_test.py", line 453, in setUp
    surface = pygame.display.set_mode(size, flags, depth)
pygame.error

======================================================================
ERROR: test_line__color (pygame.tests.draw_test.DrawLineTest)
Tests if the line drawn is the correct color.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "X:\Python37\lib\site-packages\pygame\tests\draw_test.py", line 453, in setUp
    surface = pygame.display.set_mode(size, flags, depth)
pygame.error

======================================================================
ERROR: test_line__gaps (pygame.tests.draw_test.DrawLineTest)
Tests if the line drawn contains any gaps.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "X:\Python37\lib\site-packages\pygame\tests\draw_test.py", line 453, in setUp
    surface = pygame.display.set_mode(size, flags, depth)
pygame.error

======================================================================
ERROR: test_line_endianness (pygame.tests.draw_test.DrawLineTest)
test color component order
----------------------------------------------------------------------
Traceback (most recent call last):
  File "X:\Python37\lib\site-packages\pygame\tests\draw_test.py", line 453, in setUp
    surface = pygame.display.set_mode(size, flags, depth)
pygame.error

======================================================================
ERROR: test_lines__color (pygame.tests.draw_test.DrawLineTest)
Tests if the lines drawn are the correct color.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "X:\Python37\lib\site-packages\pygame\tests\draw_test.py", line 453, in setUp
    surface = pygame.display.set_mode(size, flags, depth)
pygame.error

======================================================================
ERROR: test_lines__gaps (pygame.tests.draw_test.DrawLineTest)
Tests if the lines drawn contain any gaps.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "X:\Python37\lib\site-packages\pygame\tests\draw_test.py", line 453, in setUp
    surface = pygame.display.set_mode(size, flags, depth)
pygame.error

======================================================================
ERROR: test_path_data_validation (pygame.tests.draw_test.DrawLineTest)
Test validation of multi-point drawing methods.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "X:\Python37\lib\site-packages\pygame\tests\draw_test.py", line 453, in setUp
    surface = pygame.display.set_mode(size, flags, depth)
pygame.error

======================================================================
ERROR: test_aaline__color (pygame.tests.draw_test.PythonDrawLineTest)
Tests if the aaline drawn is the correct color.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "X:\Python37\lib\site-packages\pygame\tests\draw_test.py", line 453, in setUp
    surface = pygame.display.set_mode(size, flags, depth)
pygame.error

======================================================================
ERROR: test_aaline__gaps (pygame.tests.draw_test.PythonDrawLineTest)
Tests if the aaline drawn contains any gaps.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "X:\Python37\lib\site-packages\pygame\tests\draw_test.py", line 453, in setUp
    surface = pygame.display.set_mode(size, flags, depth)
pygame.error

======================================================================
ERROR: test_aalines__color (pygame.tests.draw_test.PythonDrawLineTest)
Tests if the aalines drawn are the correct color.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "X:\Python37\lib\site-packages\pygame\tests\draw_test.py", line 453, in setUp
    surface = pygame.display.set_mode(size, flags, depth)
pygame.error

======================================================================
ERROR: test_aalines__gaps (pygame.tests.draw_test.PythonDrawLineTest)
Tests if the aalines drawn contain any gaps.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "X:\Python37\lib\site-packages\pygame\tests\draw_test.py", line 453, in setUp
    surface = pygame.display.set_mode(size, flags, depth)
pygame.error

======================================================================
ERROR: test_line__color (pygame.tests.draw_test.PythonDrawLineTest)
Tests if the line drawn is the correct color.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "X:\Python37\lib\site-packages\pygame\tests\draw_test.py", line 453, in setUp
    surface = pygame.display.set_mode(size, flags, depth)
pygame.error

======================================================================
ERROR: test_line__gaps (pygame.tests.draw_test.PythonDrawLineTest)
Tests if the line drawn contains any gaps.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "X:\Python37\lib\site-packages\pygame\tests\draw_test.py", line 453, in setUp
    surface = pygame.display.set_mode(size, flags, depth)
pygame.error

======================================================================
ERROR: test_lines__color (pygame.tests.draw_test.PythonDrawLineTest)
Tests if the lines drawn are the correct color.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "X:\Python37\lib\site-packages\pygame\tests\draw_test.py", line 453, in setUp
    surface = pygame.display.set_mode(size, flags, depth)
pygame.error

======================================================================
ERROR: test_lines__gaps (pygame.tests.draw_test.PythonDrawLineTest)
Tests if the lines drawn contain any gaps.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "X:\Python37\lib\site-packages\pygame\tests\draw_test.py", line 453, in setUp
    surface = pygame.display.set_mode(size, flags, depth)
pygame.error

----------------------------------------------------------------------
Ran 1008 tests in 29.744s

FAILED (errors=20)
```

**Comment by illume** [comment]
Hello!

Thanks for the report :)

I guess this works ok on that windows machine with the pre-compiled binaries (from pypi)?
May be a SDL-tip issue, unless the API has changed in the 1.2 tip. I expect SDL1.2 tip doesn't have all that much real world testing done on it.

My current plan is to try to complete SDL2 support, rather than try and continue with SDL1.2 things.

**Comment by cgohlke** [comment]
This seems to have been fixed in SDL-1.2.16. I don't get any test errors with current SDL-1.2.tip (e7ee29d04712).
